CI: Enable go linter in GH Actions

Enable go linter as a GH Actions workflow under `static.yaml` (Similar to - https://github.com/kubernetes-csi/csi-driver-smb/blob/master/.github/workflows/static.yaml). However, the check will still fail as the code in this repository has many linting errors which need to be fixed.

I hit that too:
```
# hack/verify-all.sh
Verifying gofmt
No issue found
Verifying govet
go: downloading k8s.io/api v0.18.0
go: downloading k8s.io/kubernetes v1.18.0
go: downloading k8s.io/client-go v0.18.0
go: downloading k8s.io/component-base v0.18.0
go: downloading k8s.io/apiserver v0.18.0
go: downloading k8s.io/cloud-provider v0.18.0
go: downloading k8s.io/kubectl v0.18.0
go: downloading github.com/prometheus/client_golang v1.5.1
go: downloading k8s.io/cri-api v0.18.11-rc.0
go: downloading k8s.io/kube-scheduler v0.18.0
go: downloading k8s.io/csi-translation-lib v0.18.0
Done
Verifying golint
pkg/nfs/controllerserver.go:90:6: `validCapacity` is unused (deadcode)
func validCapacity(requested int64, volumePath string) error {
     ^
test/e2e/testsuites/testsuites.go:54:2: `pollLongTimeout` is unused (deadcode)
        pollLongTimeout            = 5 * time.Minute
        ^
test/e2e/e2e_suite_test.go:23:2: `reportDirEnv` is unused (deadcode)
        reportDirEnv     = "ARTIFACTS"
        ^
test/e2e/e2e_suite_test.go:24:2: `defaultReportDir` is unused (deadcode)
        defaultReportDir = "test/e2e"
        ^
pkg/nfs/server.go:94:14: Error return value of `server.Serve` is not checked (errcheck)
        server.Serve(listener)
                    ^
cmd/nfsplugin/main.go:37:10: Error return value of `flag.Set` is not checked (errcheck)
        flag.Set("logtostderr", "true")
                ^
cmd/nfsplugin/main.go:42:24: Error return value of `flag.CommandLine.Parse` is not checked (errcheck)
        flag.CommandLine.Parse([]string{})
                              ^
cmd/nfsplugin/main.go:55:32: Error return value of `cmd.MarkPersistentFlagRequired` is not checked (errcheck)
        cmd.MarkPersistentFlagRequired("nodeid")
                                      ^
cmd/nfsplugin/main.go:58:32: Error return value of `cmd.MarkPersistentFlagRequired` is not checked (errcheck)
        cmd.MarkPersistentFlagRequired("endpoint")
                                      ^
cmd/nfsplugin/main.go:62:16: Error return value of `cmd.ParseFlags` is not checked (errcheck)
        cmd.ParseFlags(os.Args[1:])
                      ^
pkg/nfs/controllerserver.go:108:2: var `volumeId` should be `volumeID` (golint)
        volumeId := req.GetVolumeId()
        ^
pkg/nfs/controllerserver.go:345:29: method getVolumeIdFromNfsVol should be getVolumeIDFromNfsVol (golint)
func (cs *ControllerServer) getVolumeIdFromNfsVol(vol *nfsVolume) string {
                            ^
pkg/nfs/controllerserver.go:354:29: method getNfsVolFromId should be getNfsVolFromID (golint)
func (cs *ControllerServer) getNfsVolFromId(id string) (*nfsVolume, error) {
                            ^
pkg/nfs/controllerserver_test.go:19:2: const `testVolumeId` should be `testVolumeID` (golint)
        testVolumeId  = "test-server/test-base-dir/test-csi"
        ^
pkg/nfs/nfs.go:55:58: exported func NewNFSdriver returns unexported type *github.com/kubernetes-csi/csi-driver-nfs/pkg/nfs.nfsDriver, which can be annoying to use (golint)
func NewNFSdriver(nodeID, endpoint string, perm *uint32) *nfsDriver {
                                                         ^
pkg/nfs/nfs.go:85:59: exported func NewNodeServer returns unexported type *github.com/kubernetes-csi/csi-driver-nfs/pkg/nfs.nodeServer, which can be annoying to use (golint)
func NewNodeServer(n *nfsDriver, mounter mount.Interface) *nodeServer {
                                                          ^
pkg/nfs/nodeserver.go:110:10: `if` block ends with a `return` statement, so drop this `else` and outdent its block (golint)
                } else {
                       ^
pkg/nfs/nfs.go:124:2: S1023: redundant `return` statement (gosimple)
        return
        ^
pkg/nfs/server.go:42:2: S1023: redundant `return` statement (gosimple)
        return
        ^
pkg/nfs/controllerserver_test.go:39:3: SA1006: printf-style function with dynamic format string and no further arguments should use print-style function instead (staticcheck)
                fmt.Printf(err.Error())
                ^
pkg/nfs/nodeserver.go:166:6: func `getVolumeMetrics` is unused (unused)

```

--- a/.github/workflows/static.yaml
+++ b/.github/workflows/static.yaml
@@ -4,17 +4,23 @@ on:
     push: {}
 jobs:
     go_lint:
-        # Enable once https://github.com/kubernetes-csi/csi-driver-nfs/issues/90 is fixed
-        # name: Go Lint
-        # runs-on: ubuntu-18.04
-        # steps:
-        #     - name: Checkout code
-        #       uses: actions/checkout@master
-        #     - name: Run linter
-        #       uses: golangci/golangci-lint-action@v1
-        #       with:
-        #           version: v1.29
-        #           args: -E=gofmt,golint,misspell --timeout=30m0s
+        name: Go Lint
+        runs-on: ubuntu-latest
+        steps:
+            - name: Checkout code
+              uses: actions/checkout@master
+            - name: Run linter
+              uses: golangci/golangci-lint-action@v2
+              with:
+                  version: v1.29
+                  args: -E=gofmt,golint,misspell --timeout=30m0s
+    verify-helm:
+        name: Verify Helm
+        runs-on: ubuntu-latest
+        steps:
+            - name: Checkout code
+              uses: actions/checkout@master
             - name: Verify Helm
               run: |
-                hack/verify-helm-chart.sh
+                sudo snap install yq
+                sudo hack/verify-helm-chart.sh

